pkg/controller/build/build_controller.go

```diff
@@ -39,8 +39,77 @@ const succeedStatus string = "Succeeded"
 const namespace string = "namespace"
 const name string = "name"
 
-type setOwnerReferenceFunc func(owner, object metav1.Object, scheme *runtime.Scheme) error
+// error message
+const (
+	// validate secrets
+	InitialErrorCodes = 0000*1 + iota
+	ListSecretInNamespaceFailed
+	NoSecretsInNamespace
+	SecretsDoNotExist
+	SecretDoesNotExist
+	// validate strategy
+	UnknownStrategy
+	ListBuildStrategyInNamespaceFailed
+	NoneBuildStrategyFoundInNamespace
+	BuildStrategyDoesNotExistInNamespace
+	ListClusterBuildStrategyFailed
+	NoClusterBuildStrategyFound
+	ClusterBuildStrategyDoesNotExist
+	// validate runtime
+	RuntimePathsCanNotBeEmpty
+	// validate BuildRun OwnerReferences
+	SetOwnerReferenceFailed
+)
 
+type setOwnerReferenceFunc func(owner, object metav1.Object, scheme *runtime.Scheme) error
+type ErrorCodes int
+
+// Validate if the errorCode exists or not, Convert errorCode type from int to string and add "B" prefix to represent build
+func ValidateAndGenerateBuildErrorCode(code ErrorCodes) (string, error) {
```

Comment:
we don want this validation to be part of the business logic of the Build controller. Mainly because the error codes are static once this is running into a container, so there is no need to validate something is not gonna change. This validation should belong into the Build unit-tests

Reply:
@qu1queee I add unit tests to test all error codes:https://github.com/shipwright-io/build/pull/483/files#diff-6d542a34b1dd297b08e4efbd1185a4f6c82b4b9e5132447ffb13783f8b7b5accR522-R589 pls take a look again, thanks a lot!